lib/src/dlisio.cpp

```diff
@@ -258,6 +258,29 @@ int dlis_sul( const char* xs,
     return DLIS_UNEXPECTED_VALUE;
 }
 
+int dlis_find_sul(const char* from,
+                  long long search_limit,
+                  long long* offset) {
+
+    static const auto needle = "RECORD";
+    const auto to = from + search_limit;
```

Comment:
I would suggest to add to docs that if someone passes search limit that reads outside of memory, it's their problem.

Reply:
I considered it, but's given by having a search_limit and a pointer in the interface, so I omitted it.